lib/service/Service.ts

```diff
@@ -121,17 +119,14 @@ class Service extends EventEmitter {
     const { currency } = args;
     const balances = new Map<string, { balance: number, pendingOpenBalance: number }>();
     const getBalance = async (currency: string) => {
-      const client = this.lndClients[currency.toUpperCase()]
-        // TODO: support all registered tokens for Raiden
-        || (currency.toUpperCase() === 'WETH' && this.raidenClient);
-
-      if (!client) {
-        // TODO: throw an error here indicating that lnd is disabled for this currency
+      try {
+        const client = this.swapClientsManager.get(currency.toUpperCase());
+        const channelBalance = await client.channelBalance();
+        return channelBalance;
+      } catch (e) {
```

Comment:
I think this may be better without the try/catch, because if we get an error on the call it will bubble up to the grpc layer and return a grpc error. With the try/catch, if there's an error we return balances of 0 which is not accurate. I'd leave that TODO in there (and this will remind me to fix it) or if you want you can have it throw an error rather than return 0 balances when there's no swap client.

Reply:
Nice solution!